### PR TITLE
Add electrification constraint to pathfinding

### LINKED ISSUE #1913: Adapt pathfinding to take into account electrification
## Associated Feature / User Story

pathfinding: electrification

## Description

The pathfinding should not send electric rolling stocks on paths without catenaries

## Goal

There paths could not be used in real life

## Acceptance criteria

- [x] No electric rolling stock is sent on a path without catenary

## Definition of ready

- [ ] The enabler story is well defined
- [ ] Acceptance criterias are defined
- [ ] The individual responsible for validating the enbaler story is assigned
- [ ] Does not depend on an missing feature
- [ ] Negociable with the implementation team
- [ ] Valuable to the end users
- [ ] Estimable by the implementation team
- [ ] As small as it can be
- [ ] Testable by the implementation team

## Definition of done

- [x] Acceptance criteria are met
- [x] Automated tests are written against acceptance criteria
- [x] Story ok-ed by Product Owner
- [ ] Documentation updated

### LINKED ISSUE #1914: Check & Adapt Rolling Stock model
## Associated Feature / User Story

pathfinding: electrification

## Description

Add data to rolling stock models to know which trains need which catenary

## Goal

This extra data is needed to avoids non realistic paths

## Acceptance criteria

- [x] We have enough data in the rolling stock model to add pathfinding constraints

1 / 2 / 3 / 5 / 8

## Definition of ready

- [ ] The enabler story is well defined
- [ ] Acceptance criterias are defined
- [ ] The individual responsible for validating the enbaler story is assigned
- [ ] Does not depend on an missing feature
- [ ] Negociable with the implementation team
- [ ] Valuable to the end users
- [ ] Estimable by the implementation team
- [ ] As small as it can be
- [ ] Testable by the implementation team

## Definition of done

- [x] Acceptance criteria are met
- [x] Automated tests are written against acceptance criteria
- [x] Story ok-ed by Product Owner
- [ ] Documentation updated

### PR DESCRIPTION
**Pull request by eckter**

### CODE REVIEW

File: api/osrd_infra/schemas/rolling_stock.py [hunk 1]
```diff
@@ -61,6 +61,8 @@ class RollingStock(BaseModel, extra=Extra.forbid):
     mass: PositiveFloat = Field(description="The mass of the train, in kg")
     rolling_resistance: RollingResistance = Field(description="The formula to use to compute rolling resistance")
     loading_gauge: LoadingGaugeType
+    electric_only: bool = Field(description="If true, the train can only use tracks with compatible catenaries")
+    compatible_voltages: List[int] = Field(description="A list of compatible voltage (in V)")
```
Comment (multun) [inline api/osrd_infra/schemas/rolling_stock.py:65]: that's not the only place where a schema has to change: the openapi spec is also affected

File: api/osrd_infra/models/rolling_stock.py [hunk 2]
```diff
@@ -64,6 +64,16 @@ class RollingStock(models.Model):
...
 
+    # Catenary related
+    electric_only = models.BooleanField(
+        help_text=_("If true, the train can only use tracks with compatible catenaries")
+    )
+    compatible_voltages = ArrayField(
+        models.PositiveIntegerField(),
+        blank=True,
+        help_text=_("A list of compatible voltage (in V)"),
+    )
```
Comment (flomonster) [inline api/osrd_infra/models/rolling_stock.py:75]: Do we have cases where a rolling stock has compatible voltages but isn't electric only

File: core/src/main/java/fr/sncf/osrd/infra/implementation/tracks/undirected/UndirectedInfraBuilder.java [hunk 3]
```diff
@@ -91,9 +92,26 @@ private TrackInfra parse(RJSInfra infra) {
 
         addSpeedSections(infra.speedSections, trackSectionsByID);
 
+        if (infra.catenaries != null)
```
Comment (flomonster) [inline core/src/main/java/fr/sncf/osrd/infra/implementation/tracks/undirected/UndirectedInfraBuilder.java]: `infra.caternaries` shouldn't be `null`

**Review by flomonster — CHANGES_REQUESTED** [review]
We can bump `RAILJSON_ROLLING_STOCK_VERSION`

File: api/osrd_infra/models/rolling_stock.py (hunk 2 above)
Comment (eckter) [reply api/osrd_infra/models/rolling_stock.py:75]: Yes, for some rolling stocks they can use electricity but it's not their only mode. They can go everywhere but eventually we'll need the compatible voltages to pick the correct effort curve.

**Review by flomonster — APPROVED** [review]

**Pull request merged**